src/core_plugins/kibana/server/lib/crypto.js

```diff
@@ -0,0 +1,106 @@
+import crypto from 'crypto';
```

Comment:
It might be nice to add some docs at the top explaining the use cases this lib is appropriate/designed for, and which it's inappropriate for (e.g. hashing sensitive info like a password before saving it in a DB).

Shaunuk's PR description also seems like a great starting point:

```javascript
/**
 * This crypto module can be used for easily encrypting and decrypting serializable objects. The
 * ease-of-use comes from the fact that this module is opinionated in its (strong) choice of
 * cryptographic algorithms, lengths, and iterations that cannot be overriden by its users.
 *
 * The module takes as input an encryption key and provides in return two functions, `encrypt` and
 * `decrypt`, which return Promises and are inverses of each other. That is:
 *
 * `someSerializableObj === await decrypt(await encrypt(someSerializableObj));
 *
 * Further, this module is exposed by the Kibana plugin for other plugins to use like so:
 *
 * const encryptionKey = 'some encryption key';
 * const { encrypt, decrypt } = server.plugins.kibana.crypto({
 *   encryptionKey
 * });
 */
```